File: design-patterns/go/factory/transport-method/helicopter.go
// helicopter.go

package main

import "fmt"

type Helicopter struct {
}

func NewHelicopter() (helicopter *Helicopter) {
	helicopter = &Helicopter{}
	return
}

func (helicopter *Helicopter) Repair() {
	fmt.Println("Helicopter Repair")
}

func (helicopter *Helicopter) Start() {
	fmt.Println("Helicopter started")
}

func (helicopter *Helicopter) Stop() {
	fmt.Println("Helicopter Stopped")
}

File: design-patterns/php/factory/transport-method/Helicopter.php
<?php
// Helicopter.php

namespace Factory\TransportMethod;

class Helicopter implements Transport {
    public function start() {
        echo "Helicopter started\n";
    }

    public function stop() {
        echo "Helicopter Stopped\n";
    }

    public function repair() {
        echo "Helicopter Repair\n";
    }
}